### PR TITLE
Addition of HTPC Manager

### PR DESCRIPTION
**Pull request by sytone**
Addition of HTPC Manager to make management better of different HTPC tools. 

Reversed out the process for a new app, hopefully it is correct! :)

### CODE REVIEW

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -1076,6 +1076,16 @@ _EOF_
 					  aSOFTWARE_TYPE[$index_current]=0
 			 aSOFTWARE_ONLINEDOC_URL[$index_current]='f=8&t=5&p=7575#p7575'
 
+		#------------------
+		index_current=155
```
Comment (Fourdee) [inline dietpi/dietpi-software:1080]: This is the index assigned to the software title, it must be used for all entries of:
```INSTALLING_INDEX=```

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -6649,6 +6659,34 @@ _EOF_
 
 		fi
 
+		#HTPC Manager
+		INSTALLING_INDEX=152
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Needs to match original index:
```INSTALLING_INDEX=155```

File: dietpi/dietpi-software [hunk 3]
```diff
@@ -6649,6 +6659,34 @@ _EOF_
...
+			INSTALL_URL_ADDRESS='https://github.com/Hellowlol/HTPC-Manager.git'
+
+			/DietPi/dietpi/func/check_connection "$INSTALL_URL_ADDRESS"
+			#Install
+			if (( $? == 0 )); then
+
+				#Install Python and PIP
+				AGI python python-pip python-imaging
+
+				cd "$HOME"
```
Comment (Fourdee) [inline dietpi/dietpi-software:6677]: Not required as this is called before the start of all software installs, however, no harm.

File: dietpi/dietpi-software [hunk 4]
```diff
@@ -6704,6 +6742,36 @@ _EOF_
 
 		fi
 
+
+		#HTPC Manager
+		INSTALLING_INDEX=200
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Needs to match original index:
```INSTALLING_INDEX=155```

File: dietpi/dietpi-software [hunk 5]
```diff
@@ -11022,6 +11090,27 @@ _EOF_
 
 		fi
 
+		#HTPC Manager
+		INSTALLING_INDEX=152
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Needs to match original index:
```INSTALLING_INDEX=155```

File: dietpi/dietpi-software [hunk 6]
```diff
@@ -11061,6 +11150,27 @@ WantedBy=multi-user.target
 _EOF_
 
 		fi
+
+		#HTPC Manager
+		INSTALLING_INDEX=200
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Needs to match original index:
```INSTALLING_INDEX=155```

File: dietpi/dietpi-software [hunk 7]
```diff
@@ -11484,6 +11594,14 @@ _EOF_
 			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/nzbget
 			rm /etc/systemd/system/nzbget.service
 
+		elif (( $1 == 152 )); then
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Needs to match original index:
```INSTALLING_INDEX=155```

eg:
```
elif (( $1 == 155 )); then
```

File: dietpi/dietpi-software [hunk 8]
```diff
@@ -11022,6 +11090,27 @@ _EOF_
...
+After=network.target
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/python $FP_DIETPI_USERDATA_DIRECTORY/htpc-manager/Htpc.py
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+			systemctl daemon-reload
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Not required as both these entries are called at the end of software installations, however, no harm.

File: dietpi/dietpi-software [hunk 9]
```diff
@@ -11061,6 +11150,27 @@ WantedBy=multi-user.target
 _EOF_
 
 		fi
+
+		#HTPC Manager
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Duplicate code at line 11093.

File: dietpi/dietpi-software [hunk 10]
```diff
@@ -6704,6 +6742,36 @@ _EOF_
 
 		fi
 
+
+		#HTPC Manager
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Duplicate code at line 6662

**Comment by Fourdee** [comment]
@sytone 

Good effort 👍 

There are a few issues with index number assigned to the installation, needs to match ```155``` throughout. Also, appears to be duplicate install and configuration sections (2 of each).

I've made comments at the sections that need changes:
https://github.com/Fourdee/DietPi/pull/1083/files

**Comment by Fourdee** [comment]
@sytone 
Also, we need to increase the max array size by 1 for software titles to ```156```, this will ensure your entry ```155``` exists:
```TOTAL_SOFTWARE_INDEXS_HARDLIMIT=156``` inside ```dietpi-software```

File: dietpi/dietpi-software [hunk 11]
```diff
@@ -11484,6 +11594,14 @@ _EOF_
 			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/nzbget
 			rm /etc/systemd/system/nzbget.service
 
+		elif (( $1 == 152 )); then
+
+			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/htpc-manager
+			
+			systemctl disable htpc-manager.service
```
Comment (Fourdee) [inline dietpi/dietpi-software]: ```systemctl disable htpc-manager.service	```
Not required, as ```dietpi-services``` will disable this service during install (to take control), however, no harm.

File: dietpi/dietpi-software [hunk 12]
```diff
@@ -11484,6 +11594,14 @@ _EOF_
 			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/nzbget
 			rm /etc/systemd/system/nzbget.service
 
+		elif (( $1 == 152 )); then
+
+			rm -R "$FP_DIETPI_USERDATA_DIRECTORY"/htpc-manager
+			
+			systemctl disable htpc-manager.service			
+			rm /etc/systemd/system/htpc-manager.service
+			systemctl daemon-reload
```
Comment (Fourdee) [inline dietpi/dietpi-software]: ```systemctl daemon-reload```
Not required, as ```systemctl daemon-reload``` is called at the end of uninstall script. However, no harm.

File: dietpi/dietpi-software (hunk 1 above)
Comment (sytone) [reply dietpi/dietpi-software:1080]: Should be 155 everywhere. It was 152 when I was testing. Fixing for next iteration.

File: dietpi/dietpi-software (hunk 2 above)
Comment (sytone) [reply dietpi/dietpi-software]: Fixed.

File: dietpi/dietpi-software (hunk 3 above)
Comment (sytone) [reply dietpi/dietpi-software:6677]: The review is not showing the line not required. is it banner?

File: dietpi/dietpi-software (hunk 10 above)
Comment (sytone) [reply dietpi/dietpi-software]: Removed

File: dietpi/dietpi-software (hunk 5 above)
Comment (sytone) [reply dietpi/dietpi-software]: Fixed.

File: dietpi/dietpi-software (hunk 8 above)
Comment (sytone) [reply dietpi/dietpi-software]: Removed.

File: dietpi/dietpi-software (hunk 9 above)
Comment (sytone) [reply dietpi/dietpi-software]: Removed

File: dietpi/dietpi-software (hunk 6 above)
Comment (sytone) [reply dietpi/dietpi-software]: removed.

File: dietpi/dietpi-software (hunk 11 above)
Comment (sytone) [reply dietpi/dietpi-software]: Removed.

File: dietpi/dietpi-software (hunk 12 above)
Comment (sytone) [reply dietpi/dietpi-software]: Removed.

**Comment by sytone** [comment]
@Fourdee fixed the issues you highlighted, let me know if there are any other problems. I made a issue with a suggestion to separate these out to separate files to make this process simpler given the amount in the file now.

**Comment by Fourdee** [comment]
@sytone 

>  fixed the issues you highlighted

Great work, looks good 👍 

We'll get this merged, i'll run a test in the next few days.

File: dietpi/dietpi-software (hunk 3 above)
Comment (Fourdee) [reply dietpi/dietpi-software:6677]: @sytone 

```
cd "$HOME"
```

Although, no harm.

**Review by Fourdee — COMMENTED** [review]
Completed.

**Comment by sytone** [comment]
Thanks, when I have my account approved on the board I will do a write up for it.

**Comment by Fourdee** [comment]
@sytone 

> Thanks, when I have my account approved on the board I will do a write up for it.

Done

**Comment by Fourdee** [comment]
- Online Doc: http://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=8043#p8043
http://ip.address:8085

Notes:
- ```python-dev``` required to compile ```pip install psutil```, which inturn needed for system status.

**Pull request merged**